pkg/planner/cardinality/row_count_index.go

```diff
@@ -627,3 +633,16 @@ func getOrdinalOfRangeCond(sc *stmtctx.StatementContext, ran *ranger.Range) int
 	}
 	return len(ran.LowVal)
 }
+
+// hasColumnStats checks if we have collected stats on any of the given columns.
+func hasColumnStats(sctx planctx.PlanContext, coll *statistics.HistColl, idxCols []*expression.Column) bool {
+	if idxCols == nil {
+		return false
+	}
+	for i := range idxCols {
+		if !statistics.ColumnStatsIsInvalid(coll.GetCol(idxCols[i].UniqueID), sctx, coll, idxCols[i].UniqueID) {
```

Comment:
!ColumnStatsIsInvalid => return true means 
once we got any index column' stats, the partial check is passed? not all index columns?